src/wallet/wallet.cpp

```diff
@@ -945,9 +945,16 @@ bool CWallet::LoadUnifiedCaches()
     }
 
     // Restore decrypted Orchard notes.
-    for (const auto& [txid, walletTx] : mapWallet) {
+    for (const auto& [_, walletTx] : mapWallet) {
         if (!walletTx.mapOrchardActionData.empty()) {
-            if (!orchardWallet.RestoreDecryptedNotes(walletTx, walletTx.mapOrchardActionData)) {
+            const CBlockIndex* pTxIndex;
+            std::optional<int> blockHeight;
+            if (walletTx.GetDepthInMainChain(pTxIndex) > 0) {
```

Comment:
This method requires a lock on `cs_main`. I verified that we take this lock in `CWallet::LoadWallet`.

Reply:
Please add the corresponding `AssertLockHeld`.